miso-web/src/main/webapp/scripts/list_poolelement.js

```diff
@@ -71,104 +71,107 @@ ListTarget.poolelement = {
     return [];
   },
   createColumns: function(config, projectId) {
-    return [{
-      'sTitle': 'Dilution Name',
-      'mData': 'name',
-      'include': true,
-      'iSortPriority': 1
-    }, {
-      "sTitle": "Warnings",
-      "mData": null,
-      "mRender": WarningTarget.poolelement.tableWarnings(config.duplicateIndicesSequences, config.nearDuplicateIndicesSequences),
-      "include": true,
-      "iSortPriority": 0,
-      "bVisible": true,
-      "bSortable": false
-    }, {
-      'sTitle': 'Proportion',
-      'sType': 'numeric',
-      'mData': 'proportion',
-      'include': !config.add,
-      'iSortPriority': 0
-    }, ListUtils.idHyperlinkColumn("Library Name", "library", "library.id", function(dilution) {
-      return dilution.library.name;
-    }, 0, true, "noPrint"), ListUtils.labelHyperlinkColumn("Library Alias", "library", function(dilution) {
-      return dilution.library.id;
-    }, "library.alias", 0, true), ListUtils.idHyperlinkColumn("Sample Name", "sample", "library.parentSampleId", function(dilution) {
-      return "SAM" + dilution.library.parentSampleId;
-    }, 0, true, "noPrint"), ListUtils.labelHyperlinkColumn("Sample Alias", "sample", function(dilution) {
-      return dilution.library.parentSampleId;
-    }, "library.parentSampleAlias", 0, true, "noPrint"), {
-      'sTitle': 'Conc.',
-      'sType': 'numeric',
-      'mData': 'concentration',
-      'include': true,
-      'iSortPriority': 0,
-      'mRender': function(data, type, full) {
-        if (type === 'display' && !!data) {
-          var units = Constants.concentrationUnits.find(function(unit) {
-            return unit.name == full.concentrationUnits;
-          });
-          if (!!units) {
-            return data + ' ' + units.units;
+    return [
+        {
+          'sTitle': 'Dilution Name',
+          'mData': 'name',
+          'include': true,
+          'iSortPriority': 1
+        },
+        {
+          "sTitle": "Warnings",
+          "mData": null,
+          "mRender": Warning.tableWarningRenderer(WarningTarget.poolelement.makeTarget(config.duplicateIndicesSequences,
+              config.nearDuplicateIndicesSequences)),
+          "include": true,
+          "iSortPriority": 0,
+          "bVisible": true,
+          "bSortable": false
+        }, {
+          'sTitle': 'Proportion',
+          'sType': 'numeric',
+          'mData': 'proportion',
+          'include': !config.add,
+          'iSortPriority': 0
+        }, ListUtils.idHyperlinkColumn("Library Name", "library", "library.id", function(dilution) {
+          return dilution.library.name;
+        }, 0, true, "noPrint"), ListUtils.labelHyperlinkColumn("Library Alias", "library", function(dilution) {
```

Comment:
The formatting here makes it harder to see at a glance what is going in, because it look like the close-braces is the end of the column - but the end of the column is the close-parentheses after "noPrint". It might be more readable if the "return....}" lines of the inner functions were indented more? Or if each ListUtils.X was on a new line?

Reply:
The JavaScript formatter in Eclipse is terrible. You are welcome to fight with it if you'd like :grin: